conans/model/scm.py

```diff
@@ -46,7 +46,7 @@ def __repr__(self):
         d = {"url": self.url, "revision": self.revision, "username": self.username,
              "password": self.password, "type": self.type, "verify_ssl": self.verify_ssl,
              "subfolder": self.subfolder, "submodule": self.submodule}
-        d = {k: v for k, v in d.items() if v}
+        d = {k: v for k, v in d.items() if v is not None and v is not ""}
```

Comment:
I am still not sure about removing empty strings. If someone defines an empty string, it will be removed from the dict, and it will become None. What tests were failing? Maybe the tests need to be fixed?